services/skus/credentials.go

```diff
@@ -673,3 +649,37 @@ func (s *Service) DeleteOrderCreds(ctx context.Context, orderID uuid.UUID, isSig
 
 	return nil
 }
+
+// checkNumBlindedCreds checks the number of submitted blinded credentials.
+//
+// The number of submitted credentials must not exceed:
+// - for single-use the quantity of the item;
+// - for time-limited-v2 the product of numPerInterval and numIntervals.
+func checkNumBlindedCreds(ord *model.Order, item *model.OrderItem, ncreds int) error {
+	switch item.CredentialType {
+	case singleUse:
+		if ncreds > item.Quantity {
+			return errInvalidNCredsSingleUse
+		}
+
+		return nil
+	case timeLimitedV2:
+		nperInterval, err := ord.NumPerInterval()
+		if err != nil {
+			return err
+		}
+
+		nintervals, err := ord.NumIntervals()
+		if err != nil {
+			return err
+		}
+
+		if ncreds > nperInterval*nintervals {
+			return errInvalidNCredsTlv2
+		}
+
+		return nil
+	default:
+		return nil
```

Comment:
question: is it safe to return nil if we receive an unknown credential type?

Reply:
Good question. This refactoring is based on the existing assumptions.

I would expect the caller to take care about that, and this code can be left for now. However, we can also return a sentinel error and handle it. But this would complicate handling of that error at the call site.

What do you think?